### PR TITLE
Show stats for string & date fields in stats dock

### LINKED ISSUE #1018: [FEATURE] Show stats for string and date fields in stats dock
Original commit: https://github.com/qgis/QGIS/commit/88d3e2aa7f88454d544c8e3a960a30bac33ed21c by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by yjacolin**
Fix #1018

### CODE REVIEW

File: source/docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -292,6 +292,27 @@ allows users to choose:
 * refresh the informations;
 * the statistics information to display with the bottom right button.
 
+Statistic information available are (in regards of the field type):
+
+* Sum (float or integer)
+* Mean (float or integer)
+* Median (float or integer)
+* Standard Deviation (float or integer)
+* Standard Deviation on Sample (float or integer)
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: on sample?

File: source/docs/user_manual/introduction/general_tools.rst [hunk 2]
```diff
@@ -292,6 +292,27 @@ allows users to choose:
...
+* Minimal value (date, string, float or integer)
+* Maximal value (date, string, float or integer)
+* Range (date)
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: not only. numeric values also have this option available

File: source/docs/user_manual/introduction/general_tools.rst [hunk 3]
```diff
@@ -292,6 +292,27 @@ allows users to choose:
 * refresh the informations;
 * the statistics information to display with the bottom right button.
 
+Statistic information available are (in regards of the field type):
```
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/general_tools.rst]: Statistic information available are (depending on the field's type):

File: source/docs/user_manual/introduction/general_tools.rst [hunk 4]
```diff
@@ -292,6 +292,27 @@ allows users to choose:
...
+* Variety (float or integer)
+* First Quartile (float or integer)
+* Third Quartile (float or integer)
+* Inter Quartile Range (float or integer)
+* Count Distinct Value (date or string)
+* Count number of Missing Value (date or string)
+* Minimum Length (string)
+* Maximum Length (string)
+
 .. note:: The list of shown statistics depends on the type of the values (numeric,
```
Comment (SrNetoChan) [inline source/docs/user_manual/introduction/general_tools.rst]: I would say that this note is duplicated now since it is explained in the top of the list.

**Comment by SrNetoChan** [comment]
Apart from the to comments, LGTM.

**Pull request merged**